DOCS: Initial Framework grammar and tone edits (#181)
wpengine/faustjs: docs/templating/README.md

## Template Hierarchy in Next.js

> While Next.js has a robust [routing layer](https://nextjs.org/docs/routing/introduction), it is not always ideal for WordPress-based sites routes are built around directory and file structures rather than content types.

Reviewer: ```suggestion
While Next.js has a robust [routing layer](https://nextjs.org/docs/routing/introduction), it is not always ideal for WordPress-based sites where routes are built around directory and file structures rather than content types.
```

> ): Promise<void>;

Reviewer 2: []()

```

> These functions can be used to make additional requests using the Apollo Client that will be executed when Next.js [Data Fetchers](https://nextjs.org/docs/basic-features/data-fetching) run.

Reviewer 2: []()

> You may find this useful when you need additional data from the WordPress backend that doesn't fetch by default.

Reviewer: ```suggestion
You may find this useful when you need additional data from the WordPress backend that is not fetched by default.
```

Reviewer 3: I am going to change this back to my original commit since "not fetched" is in a passive voice.

> You may find this helpful when you need additional data from the WordPress backend that does not fetch by default.

Reviewer: @DonNateR How do you feel about this?

```suggestion
You may find this helpful when you need additional data from the WordPress backend that is not present by default.
```

Or just:

```suggestion
You may find this helpful when you need additional data from the WordPress backend.
```

`does not fetch` reads strangely in this context because it's the request that is fetching the data and not the WordPress backend.

Thank you for all your time reviewing our copy! I like the improvements here.

In an average Next.js page, you would only implement one of these functions. Since theme components are not Next.js pages, these functions call on the framework, not Next.js. If you are shipping a reusable Theme component you will want to export both functions. The framework will call on the function that corresponds to what the Next.js page uses. So if the Next.js page is using `getStaticProps`, the framework will call the `getStaticProps` function on your theme component.

### Example

> See [`single.tsx`](https://github.com/wpengine/headless-framework/blob/canary/examples/preview/theme/single.tsx) in our preview example. This template exports the component `getStaticProps` and `getServerSideProps` to modify the default data requested in the Next.js Data Fetchers.

Reviewer 2: []()

### Example

See [`single.tsx`](https://github.com/wpengine/headless-framework/blob/canary/examples/preview/theme/single.tsx) in our preview example. This template exports the component, `getStaticProps`, and `getServerSideProps` to modify the default data requested in the Next.js Data Fetchers.
See [`single.tsx`](https://github.com/wpengine/headless-framework/blob/canary/examples/preview/theme/single.tsx) in our preview example. This template exports the component `getStaticProps` and `getServerSideProps` to modify the default data requested in the Next.js Data Fetchers.

> # CMS-Based Routing (Template Hierarchy) Flow

Reviewer 2: []()
